File: Source/Bifrost.JavaScript.Specs/utils/for_StringMapping/when_matching_a_string_with_wildcard_in_the_middle.js
describe("when matching a string with wildcard in the middle", function () {
    var mapping = Bifrost.utils.StringMapping.create({
        format: "{something}/**/for_{else}",
        mappedFormat: "{something}.**.for_{else}"
    });
    var result = mapping.matches("this/is/a/wildcard/uri/for_things");

    it("should match", function () {
        expect(result).toBe(true);
    });
});

File: Source/Bifrost.Specs/Utils/for_StringMapping/when_matching_a_string_with_wildcard_in_the_middle.cs
using Machine.Specifications;

namespace Bifrost.Utils.for_StringMapping   
{
    public class when_matching_a_string_with_wildcard_in_the_middle
    {
        static StringMapping mapping = new StringMapping(
                "{something}/**/for_{else}",
                "{something}.**.for_{else}"
            );

        static bool result;

        Because of = () => result = mapping.Matches("this/is/a/wildcard/string/for_things");

        It should_match = () => result.ShouldBeTrue();
    }
}
